Cesium3DTilesSelection/src/TilesetContentManager.cpp

```diff
@@ -1146,24 +1195,39 @@ void TilesetContentManager::updateContentLoadedState(
     if (myGeometricError >= parentGeometricError) {
       tile.setUnconditionallyRefine();
     }
-  } else if (content.isRenderContent()) {
-    TileRenderContent* pRenderContent = content.getRenderContent();
-    // add copyright
-    pRenderContent->setCredits(GltfUtilities::parseGltfCopyright(
-        *this->_externals.pCreditSystem,
-        pRenderContent->getModel(),
-        tilesetOptions.showCreditsOnScreen));
-
-    // create render resources in the main thread
-    void* pWorkerRenderResources = pRenderContent->getRenderResources();
-    void* pMainThreadRenderResources =
-        this->_externals.pPrepareRendererResources->prepareInMainThread(
-            tile,
-            pWorkerRenderResources);
-    pRenderContent->setRenderResources(pMainThreadRenderResources);
+
+    tile.setState(TileLoadState::Done);
   }
+}
+
+void TilesetContentManager::finishLoading(
+    Tile& tile,
+    const TilesetOptions& tilesetOptions) {
+  // Run the main thread part of loading.
+  TileContent& content = tile.getContent();
+  TileRenderContent* pRenderContent = content.getRenderContent();
+
+  assert(pRenderContent != nullptr);
+
+  // add copyright
+  pRenderContent->setCredits(GltfUtilities::parseGltfCopyright(
+      *_externals.pCreditSystem,
+      pRenderContent->getModel(),
+      tilesetOptions.showCreditsOnScreen));
 
+  void* pWorkerRenderResources = pRenderContent->getRenderResources();
+  void* pMainThreadRenderResources =
+      _externals.pPrepareRendererResources->prepareInMainThread(
```

Comment:
`this->` was removed here too.

Reply:
Sorry, I was starting to get used to some other conventions from non-Cesium code. I was starting to think the underscore and "this->" together were redundant since I _thought_ an underscore implies a private member. But in fact, the accessibility is unrelated to member-ship, since private static variables are possible!